### PR TITLE
Improved guide to testing/contributing

### PR DESCRIPTION
**Pull request by riichard**
- Contributing guide missed the `npm build` step when setting up the repository. 
- Added the suggestion to use PHP's built in webserver for testing, next to setting up with the LAMP. PHP is by default installed on OSX.

### CODE REVIEW

**Comment by nathanhammond** [comment]
Heads up: `php -S` didn't work for me when I was setting it up just a few days ago. (Mac OS X, 10.10)

**Comment by riichard** [comment]
Which port did you use?

File: README.md [hunk 1]
```diff
@@ -173,11 +173,15 @@ grunt watch
 Run the unit tests with a local server that supports PHP. Ensure that you run the site from the root directory, not the "test" directory. No database is required. Pre-configured php local servers are available for Windows and Mac. Here are some options:
 
 - Windows: [WAMP download](http://www.wampserver.com/en/)
-- Mac: [MAMP download](http://www.mamp.info/en/index.html)
 - Linux: [Setting up LAMP](https://www.linux.com/learn/tutorials/288158-easy-lamp-server-installation)
 - [Mongoose (most platforms)](http://code.google.com/p/mongoose/)
+- Mac: [MAMP download](http://www.mamp.info/en/index.html)
```
Comment (mzgol) [inline README.md]: Why this change?

Comment (riichard) [inline README.md]: I shouldn't have committed this change. There is nothing different compared
to before.

I initially thought about altering the Mac installation description for
Lamp, because PHP was already installed on OS X.

Then I thought it would be better just to insert a quick tip for using PHPs
build server if you're on OS X or already have OS X installed.
Because maybe some people still want to use LAMP on OS X.

Comment (markelog) [inline README.md]: I'm not sure we still recommend to use those, for Windows, sure, but for the Linux or Mac? If it is a Linux user, their definitely knows who to install php and http-server. If it is a Mac user, their already have apache and php in the system.

Comment (mzgol) [inline README.md]: They still need to be configured, even on a Mac and it's not that easy.

The best solution would be to fix #1755 but it needs some effort.

File: README.md [hunk 2]
```diff
@@ -173,11 +173,15 @@ grunt watch
...
 
+If you already have PHP installed, you can use it's built in webserver to complete the tests:
 
+```
+php -S localhost:8080
```
Comment (mzgol) [inline README.md]: I've checked locally but such a server doesn't seem to work well. It failed the "core: document ready when jQuery loaded asynchronously (#13655)" test and times out/fails a lot of AJAX tests. I guess a full Apache server might be needed to make it work correctly.

Comment (gibson042) [inline README.md]: > It failed the "core: document ready when jQuery loaded asynchronously (#13655)" test and times out/fails a lot of AJAX tests.

Confirmed. Accordingly, I don't think we can accept a suggestion to use it.

**Comment by mzgol** [comment]
@riichard Did you see our comments? The standalone PHP server won't work and the Mac info was just swapped so these changes should be reverted. This will get README.md back to the state it was in, the only change will be in CONTRIBUTING.md. Would you like to apply these changes? Or do you have any other suggestions?

**Comment by riichard** [comment]
@mzgol I see your comments, the tests worked fine with php running on my machine but I guess there are exceptions and differences to apache. It's best to leave that README.md suggestion out then.

I would still like to apply the changes to CONTRIBUTING.md.

**Comment by mzgol** [comment]
> @mzgol I see your comments, the tests worked fine with php running on my machine

Did you run the whole suite? I remember I used to have a lot of issues with the standalone server, although now I see only 4 of them (I tried on OS X 10.11 GM Candidate, PHP 5.5.27):
screen shot 2015-09-11 at 20 37 51

If my memory is fine, though, it might work worse on older versions and even then the test suite still doesn't pass completely. So I think this advice would be risky in general.

> I would still like to apply the changes to CONTRIBUTING.md.

Sure. :) Just revert the other changes and we'll land this one.

**Comment by riichard** [comment]
It was a challenge for me to revert that commit from github, but I got it done. The PHP server suggestion commit is removed.

The only commit left fixes CONTRIBUTING.md's missing 'npm run build' step.

**Comment by mzgol** [comment]
LGTM.

**Comment by markelog** [comment]
LGTM
